shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -0,0 +1,647 @@
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.FlutterShellArgs;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.embedding.engine.renderer.OnFirstFrameRenderedListener;
+import io.flutter.plugin.platform.PlatformPlugin;
+import io.flutter.view.FlutterMain;
+
+import static android.content.ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW;
+
+/**
+ * Delegate that implements all Flutter logic that is the same between a {@link FlutterActivity}
+ * and a {@link FlutterFragment}.
+ * <p>
+ * <strong>Why does this class exist?</strong>
+ * <p>
+ * One might ask why an {@code Activity} and {@code Fragment} delegate needs to exist. Given
+ * that a {@code Fragment} can be placed within an {@code Activity}, it would make more sense
+ * to use a {@link FlutterFragment} within a {@link FlutterActivity}. Then, the problem of
+ * duplicated behavior is solved without extracting all of that logic into a 3rd party
+ * {@code Object}.
+ * <p>
+ * The original solution for this embedding was exactly that - a {@link FlutterFragment}
+ * embedded within a {@link FlutterActivity}. However, it was later discovered that the binary
+ * impact of including {@code Fragment}s in a Flutter app was about 100k in release mode. This
+ * binary hit would be fine for add-to-app cases because {@code Fragment}s are almost certainly
+ * being used already. However, this 100k was deemed unacceptable for full-Flutter apps, which
+ * would not otherwise have to pay that binary size cost. Therefore, it was concluded that
+ * Flutter must provide a {@link FlutterActivity} based on the AOSP {@code Activity}, and an
+ * independent {@link FlutterFragment} for add-to-app developers.
+ * <p>
+ * That is why {@code FlutterActivityAndFragmentDelegate} exists.
+ * <p>
+ * If a time ever comes where the inclusion of {@code Fragment}s in a full-Flutter app is no
+ * longer deemed an issue, this class should be immediately decomposed between
+ * {@link FlutterActivity} and {@link FlutterFragment} and then eliminated.
+ * <p>
+ * <strong>Caution when modifying this class</strong>
+ * <p>
+ * Any time that a "delegate" is created with the purpose of encapsulating the internal
+ * behaviors of another object, that delegate is highly susceptible to degeneration. It is
+ * easy to tack new responsibilities on to the delegate which would not otherwise be added
+ * to the original object. It is also easy to begin hanging listeners and callbacks on a
+ * delegate object that likewise would not be added to the original object. A delegate can
+ * quickly become a complex web of dependencies and optional references that are very
+ * difficult to track.
+ * <p>
+ * Maintainers of this class should take care to only place code in this delegate that would
+ * otherwise be placed in either {@link FlutterActivity} or {@link FlutterFragment}, and in
+ * exactly the same form. Do not use this class as a convenient shortcut for any other
```

Comment:
I'd put <strong> around this sentence which is the key sentence in this whole doc. There's a lot of text people get lost in otherwise.

Reply:
done